src/Basescape/BasescapeState.cpp

```diff
@@ -657,19 +640,21 @@ void BasescapeState::miniRightClick(Action*)
  */
 void BasescapeState::miniMiddleClick(Action *)
 {
```

Comment:
It says we can't support this yet. Why not?

Reply:
so originally bases were stored in a base vector in `SavedGame` and their position and the order was determined by their order in this vector, which this method swaps.

With `SavedGame` no longer having ownership of the data, that presents some problems. The order in `view<Base>` should be stable, so I'm using that as a substitute, but it is not trivial to modify (there is `sort<Base>` but that doesn't let me easily swap to entities positions).

I'm considering my options here. 
* There could be an `Index` component, which would be an ECS style of managing it, I guess. Or `Base` could have an `Index` property. The downside here is that managing the values of these indexes is not innate, so there could be bugs if the Indexes are not updated properly.
* SavedGame could hold a vector of base `entt::enttity` representing the order. Index order here is always correct, but since it doesn't own this data it needs to be kept in sync when the set of bases mutates which is a pain.

In both of the above this data also needs to be persisted to the save file potentially which is a minor pain.

Anyways it's not an unsolvable problem but its a fair amount of effort for a minor feature (one I didn't even know existed!) so me commenting it out is mostly just punting on it for now.

I'm leaning towards the component based solution. That will probably let me drop some helpers I wrote to query index state so... maybe I should have started with that, lol. Oh well.